Development/nmos/jwk_utils.cpp

```diff
@@ -174,101 +167,48 @@ namespace nmos
-                    // see https://tools.ietf.org/html/rfc7518#section-6.3.1.2
-                    if (!jwk.has_field(U("e"))) throw jwk_exception("jwk does not contain 'e': " + utility::us2s(jwk.serialize()));
-
-                    // using n & e to convert Json Web Key to RSA Public Key
-                    return jwk_to_public_key(jwk.at(U("n")).as_string(), jwk.at(U("e")).as_string()); // may throw jwk_exception
-                }
-                // Elliptic Curve Public Keys
-                // see https://tools.ietf.org/html/rfc7518#section-6.2
-                else if (U("EC") == jwk.at(U("kty")).as_string())
+                if (bignum)
                 {
-                    // Public Key Use (use), optional!
-                    // see https://tools.ietf.org/html/rfc7517#section-4.2
-                    if (jwk.has_field(U("use")))
+                    const auto size = BN_num_bytes(bignum);
+                    std::vector<uint8_t> data(size);
+                    if (BN_bn2bin(bignum, data.data()))
                     {
-                        if (U("sig") != jwk.at(U("use")).as_string()) throw jwk_exception("jwk contains invalid 'use': " + utility::us2s(jwk.serialize()));
+                        return utility::conversions::to_base64url(data);
                     }
-
-                    // is crv presented?
-                    // The "crv" (curve) parameter identifies the cryptographic curve used with the EC public Key
-                    // see https://tools.ietf.org/html/rfc7518#section-6.2.1.1
-                    if (!jwk.has_field(U("crv"))) throw jwk_exception("jwk does not contain 'crv': " + utility::us2s(jwk.serialize()));
-
-                    // is x presented?
-                    // Base64 URL encoded string representation of the x coordinate of the EC public Key
-                    // see https://tools.ietf.org/html/rfc7518#section-6.2.1.2
-                    if (!jwk.has_field(U("x"))) throw jwk_exception("jwk does not contains 'x': " + utility::us2s(jwk.serialize()));
-
-                    // is y presented?
-                    // Base64 URL encoded string representation of the y coordinate of the EC public Key
-                    // see https://tools.ietf.org/html/rfc7518#section-6.2.1.3
-                    if (!jwk.has_field(U("y"))) throw jwk_exception("jwk does not contain 'y': " + utility::us2s(jwk.serialize()));
-
-                    // using crv, x & y to convert Json Web Key to EC Public Key
-                    return jwk_to_public_key(jwk.at(U("crv")).as_string(), jwk.at(U("x")).as_string(), jwk.at(U("y")).as_string()); // may throw jwk_exception
-                }
-                else
-                {
-                    throw jwk_exception("jwk contains invalid 'kty': " + utility::us2s(jwk.serialize()));
                 }
-            }
+                return utility::string_t{};
+            };
```

Comment:
`extra ‘;’ [-Wpedantic]`

```suggestion
            }
```